enhancements/accelerators/out-of-tree-drivers.md

```diff
@@ -0,0 +1,539 @@
+* Day 2 are all operations involved after an OpenShift cluster is installed 
+
+In this enhancement we are solely concentrating on Day 2 operations.
+
+### DriverContainers
+
+DriverContainers are used more and more in cloud-native environments, especially when run on pure container operating systems to deliver hardware drivers to the host. Driver containers are more than a delivery mechanism for the driver itself, as they extend the kernel stack beyond the out-of-box software and hardware features of a specific kernel. Additionally, a driver container can handle the configuration of modules, and start userland services. 
+ 
+DriverContainers work on various container capable Linux distributions.
+ 
+With DriverContainers the host stays always "clean", and does not classh with different library versions or binaries on the host. Prototyping is far more easier, updates are done by pulling a new container with the loading and unloading done by the DriverContainer with checks on /proc and /sys and other files to make sure that all traces are removed).
+
+## Current Solutions
+
+Here are the current solutions in use today.
+
+### Special Resource Operator
+ 
+For any day-2 management of kernel modules we can leverage the Special Resource Operator (SRO) features. SRO was written in such a way that it is highly customizable and configurable to any hardware accelerator or out-of-tree kernel module. 
+
+A detailed description of SRO and its inner workings are described in the following two blog posts:
+
+* [https://red.ht/2JQuNwB](https://red.ht/2JQuNwB)
+* [https://red.ht/34ubzq3](https://red.ht/34ubzq3)
+ 
+SRO supports full lifecycle management of an accelerator stack but it can also be used as a stripped down version to manage e.g only one kernel module. Furthermore SRO can handle multiple kernel modules from different vendors and is able to model a dependency between those too.
+ 
+Another important feature is the ability to consume build artifacts from other kernel modules to build a more sophisticated DriverContainer. SRO is capable of delivering out-of-tree drivers and supporting software stacks for kernel features and hardware that is not shipped as part of the standard Fedora/RHEL distribution.
+
+Ideally, SRO would pull a prebuilt DriverContainer with precompiled drivers from the vendor. Any module updates (and downgrades) will be delivered by container. 
+
+The Special Resource Operator (SRO) is currently only available in OperatorHub. SRO has proven in the past to be the template for enabling hardware when on OpenShift. Its capabilities to handle several DriverContainers with only one copy of SRO running makes it a preferable solution to tackle kmods on OpenShift.
+
+SRO is going to be a core-component of OpenShift and delivered/managed by CVO. Here is an example how one can use SRO + KVC to deliver a simple kernel module via container in a OpenShift cluster: [https://bit.ly/2EAlLEF](https://bit.ly/2EAlLEF)
```

Comment:
This is a leap.

Why must SRO move under CVO management?  What problem did that solve versus its present path in OLM?

Reply:
I was not sure if a OLM based operator can prevent upgrades of a cluster. In a cluster with “special resources” which are the high assests of the cluster and not the default CPU nodes, and especailly e.g. with GPUs I do not want an upgrade if the preflight checks cannot guarantee me that the newest kernel  will break it.
22:43
As an OLM operator am I "allowed" to create an ClusterOperator object?

Reply:
> As an OLM operator am I "allowed" to create a ClusterOperator object?

Regardless of whether you get permission to create one, the CVO only cares about ClusterOperators which exist as manifests in the release image, so it wouldn't care about an OLM-child-created ClusterOperator.  For OLM-child updates, there's #592 in flight, which sounds like it might help your use-case.  And I feel like there was earlier OLM-child-update discussion somewhere else too, but I'm not finding it at the moment...

Reply:
@wking Can you please define "care", you mean even if SRO is creating the ClusterOperator object and setting the right values the CVO will not consider it as a blocker or not-blocker for an upgrade?

Reply:
@wking https://github.com/openshift/enhancements/pull/357#issuecomment-740121530 either through olm mechanisms or cluster operator object.